static/js/showPoll.js

```diff
@@ -45,3 +45,44 @@ yucibot.controller('showPoll', function($scope, $http, $log, $interval) {
   $scope.getAnswers()
   $interval($scope.getAnswers, 500);
 })
+
+// Draw pie chart
+function makeGraph() {
+  google.charts.load('current', {'packages':['corechart']});
+  setTimeout(function () {
+    google.charts.setOnLoadCallback(drawChart);
+  }, 300);
+
+  var arrChartTable = new Array;
+
+  setTimeout(function () {
+    arrChartTable.push(['Answer', 'Votes'])
+    for (x = 0; x < voted.length; x++) {
```

Comment:
Missing `var` before `x = 0;` It works in this situation, but the variable `x` is now a variable bound to the local `this` scope.

Instead create a local variable that only lives inside the function that is being called.

```javascript
for (var x = 0; x < voted.length; x++) {
  // do stuff
}
```

This for example doesn't work:

```javascript
function directAssignment() {
  var x = 0;
  
  function hello() {
    function draw() {
      for (x = 0; x < 3; x++) {
        console.log('loop', x);
      }
    }

    draw();
  }
  
  hello();

  console.log('x', x);
}

directAssignment()
// console.logs the follwing:
// loop 0
// loop 1
// loop 2
// x 3
// This means that the variable x has been changed by your loop

function reassignment() {
  var x = 0;
  
  function hello() {
    function draw() {
      for (var x = 0; x < 3; x++) {
        console.log('loop', x);
      }
    }

    draw();
  }
  
  hello();

  console.log('x', x);
}

// console.logs the follwing:
// loop 0
// loop 1
// loop 2
// x 0
// The variable x isn't changed by your loop, because it's re-created by the function.
```